pkg/controller/storagecluster/storagecluster.go

```diff
@@ -334,6 +339,78 @@ func (c *Controller) validateCustomAnnotations(current *corev1.StorageCluster) e
 	return nil
 }
 
+
+
+func getKeysFromNodeSelector(ns *corev1.NodeSelector) map[string]bool {
+	keys := make(map[string]bool)
+
+	// Use node name as selector.
+	if ns.NodeName != "" {
+		keys["NodeName"] = true
+	}
+
+	if ns.LabelSelector != nil {
+		for k, _ := range ns.LabelSelector.MatchLabels {
+			keys[k] = true
+		}
+
+		for _, r := range ns.LabelSelector.MatchExpressions {
+			keys[r.Key] = true
+		}
+	}
+
+	return keys
+}
+
+func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster) error {
+	key, err := c.getCloudStorageLabelKey(cluster)
+	if err != nil {
+		return err
+	}
+
+	if cluster.Spec.CloudStorage != nil &&
+		cluster.Spec.CloudStorage.NodePoolLabel != "" &&
+		key != "" &&
+		cluster.Spec.CloudStorage.NodePoolLabel != key {
+		return fmt.Errorf("node pool label key incorrect, expected %s, actual %s", cluster.Spec.CloudStorage.NodePoolLabel, key)
+	}
+
+	return nil
+}
+
+func (c *Controller) getCloudStorageLabelKey(cluster *corev1.StorageCluster) (string, error) {
+	if cluster.Spec.Nodes == nil || len(cluster.Spec.Nodes) == 0 {
+		return "", nil
+	}
+
+	first := false
+	key := ""
+	for _, node := range cluster.Spec.Nodes {
+		if node.CloudStorage == nil {
+			continue
+		}
+
+		keys := getKeysFromNodeSelector(&node.Selector)
+		if len(keys) != 1 {
+			return "", fmt.Errorf("it's required to have 1 key in cloud storage node label selector, key(s) %v", keys)
+		}
+
+		keyTemp := ""
+		for k, _ := range keys {
+			keyTemp = k
+		}
+
+		if !first {
```

Comment:
nit: We could just check `key == ""` instead of the `first` boolean, right?

Reply:
correct